front/src/common/redux/reducers/studentRegistrationReducer.js

```diff
@@ -0,0 +1,30 @@
+import { REGISTRATION_STUDENT, REGISTRATION_STUDENT_ERROR, HIDE_STUDENT_REGISTERED_MODAL } from '../actions/types';
+
+const initialState = {
+  firstNameEngValid: false,
+  isFormSended: false,
+  isModalVisible: false,
+  errorMessage: '',
+};
+
+export function studentRegister(state = initialState, action) {
+  const { response } = action;
+  const { error } = action;
+
+  switch (action.type) {
+    case REGISTRATION_STUDENT:
+      if (response.ok && !response.isJoi) {
+        return { ...state, isFormSended: true, isRegistered: true,  };
```

Comment:
It is better to implement it in the way, that if we get `types.LOGIN_USER_SUCCESS` action we handle success case, no error messages etc. Because if we have an error message it means that login is failed and we need to use another action